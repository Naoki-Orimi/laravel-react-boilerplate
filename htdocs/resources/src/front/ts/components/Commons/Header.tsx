import * as React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { URL } from "../../common/constants/url";

class CommonHeader extends React.Component {

  constructor(props) {
    super(props);
  }

  render(): JSX.Element {

    return (
      <React.Fragment>
      <header className="header shadow-sm">
          <nav className="navbar navbar-expand-md navbar-light bg-white headerNav">
              <a className="header_logo" href="/">
                  <img src="/assets/front/image/logo.png" alt="" className="" />
              </a>

              <div className="" id="navbarSupportedContent">
                  <ul className="navbar-nav mr-auto">

                  </ul>

                  <ul className="navbar-nav ml-auto">
                      <li className="nav-item">
                          <a className="btn btn-danger mr-3" href="/login">ログイン</a>
                      </li>
                      <li className="nav-item">
                          <a className="btn btn-link text-danger" href="/register">新規登録</a>
                      </li>
                      <li className="nav-item">
                          <a className="btn btn-link text-danger" href="#">
                              お問い合わせ
                          </a>
                      </li>

                      {/*
                      <li className="nav-item dropdown">
                          <a id="navbarDropdown" className="nav-link dropdown-toggle" href="#" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" v-pre>
                              {{ Auth::user()->name }} <span className="caret"></span>
                          </a>

                          <div className="dropdown-menu dropdown-menu-right" aria-labelledby="navbarDropdown">
                              <a className="dropdown-item" href="#" onclick="event.preventDefault();
                                                      document.getElementById('logout-form').submit();">
                                  {{ __('ログアウト') }}
                              </a>

                              <form id="logout-form" action="{{ route('logout') }}" method="POST" style="display: none;">
                                  @csrf
                              </form>

                              <a className="dropdown-item" href="#" onclick="event.preventDefault();
                                                      document.getElementById('shop-form').submit();">
                                  カートを見る
                              </a>

                              <form id="shop-form" action="{{ route('shop.mycart') }}" method="POST" style="display: none;">
                                  @csrf
                              </form>

                          </div>
                      </li>

                      <a href="#" onclick="event.preventDefault();
                                                      document.getElementById('shop-form').submit();">
                          <img src="{{ asset('/assets/front/image/cart.png') }}" className="cartImg ml-3">
                      </a>
                      <li className="nav-item">
                          <a className="btn btn-link text-danger" href="{{ route('contact.index') }}">
                              お問い合わせ
                          </a>
                      </li>

                       */}

                  </ul>
              </div>
          </nav>
      </header>
      </React.Fragment>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    parts: state.parts,
    auth: state.auth
  };
};

const mapDispatchToProps = {  };

export default connect(mapStateToProps, mapDispatchToProps)(CommonHeader);
