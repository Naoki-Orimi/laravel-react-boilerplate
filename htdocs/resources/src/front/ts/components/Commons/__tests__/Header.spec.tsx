import React from 'react';
import renderer from 'react-test-renderer';
import { CommonHeader } from '../Header';

describe('Header', () => {

  test('Header-Logout', () => {
    const component = renderer.create(<CommonHeader auth={{auth: false}} push={()=>{}}/>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  // test('Header-Login', () => {
  //   const component = renderer.create(<CommonHeader auth={{auth: true, name: 'テスト'}} push={()=>{}}/>);
  //   const tree = component.toJSON();

  //   expect(tree).toMatchSnapshot();
  // });

})
