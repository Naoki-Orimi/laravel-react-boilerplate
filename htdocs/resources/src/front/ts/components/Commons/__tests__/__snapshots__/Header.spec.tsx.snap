// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header-Logout 1`] = `
<header
  className="header shadow-sm"
>
  <nav
    className="navbar navbar-expand-md navbar-light bg-white headerNav"
  >
    <a
      className="header_logo"
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        className=""
        src="/assets/front/image/logo.png"
      />
    </a>
    <div
      className=""
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <ul
          className="navbar-nav ml-auto"
        >
          <li
            className="nav-item"
          >
            <a
              className="btn btn-danger mr-3"
              href="/login"
              onClick={[Function]}
            >
              ログイン
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="btn btn-link text-danger"
              href="/register"
              onClick={[Function]}
            >
              新規登録
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="btn btn-link text-danger"
              href="/contact"
              onClick={[Function]}
            >
              お問い合わせ
            </a>
          </li>
        </ul>
      </ul>
    </div>
  </nav>
</header>
`;
